go/private/actions/archive.bzl

```diff
@@ -38,7 +38,12 @@ def emit_archive(go, source=None):
   if go.cover:
     source, cover_vars = go.cover(go, source)
   split = split_srcs(source.srcs)
-  compilepath = source.library.importpath if source.library.importpath else source.library.name
+  if source.library.importmap:
```

Comment:
I think we should just *always* set importmap (in the library construction)
Then we could drop compilepath and just use 
```
lib_name = source.library.importmap + ".a"
```
and also not have to test it anywhere else